src/yaml.cpp

```diff
@@ -106,31 +107,49 @@ estd::result<YAML::Node, estd::error> readYamlFile(std::string const & path) {
 }
 
 YamlResult<void> mergeYamlNodes(YAML::Node & map_a, YAML::Node map_b) {
-	// Check if the arguments are maps.
-	if (!map_a.IsMap() && !map_a.IsNull()) {
-		return YamlError{"tried to merge into a YAML node that is not a map"};
-	}
-	if (!map_b.IsMap() && !map_b.IsNull()) {
-		return YamlError{"tried to merge from a YAML node that is not a map"};
-	}
-
 	for (YAML::const_iterator iterator = map_b.begin(); iterator != map_b.end(); iterator++) {
+		
 		std::string key = iterator->first.as<std::string>();
 		YAML::Node value = iterator->second;
-		if (value.IsMap() && map_a[key]) {
+	    if (value.IsMap() && map_a[key]) {
+		
 			auto merged = mergeYamlNodes(map_a[key], value);
 			if (!merged) {
 				merged.error().appendTrace({key, "", YAML::NodeType::Map});
 				return merged;
 			}
-			continue;
+			
+		}
+		// Check the tag of the Node and go through the contents if it is an ordered dictionary.
+		if (value.Tag() == "!ordered_dict") {
```

Comment:
Why check the tag on the value? I would expect you'd check the tag on `map_a` instead of each child while looping. That way, it also works when the top-level thing is an ordered dict, and you have less special cases.

You could even have `mergeYamlNodes` check it, and then delegate to `mergeYamlMaps` or `mergeYamlOrderedDicts` (or give an error). That could help keep the code easy to follow.

Reply:
As discussed on slack: it is necessary to check if the value is a map or ordered dict, because if it is not it has to be replaced rather than merged. But both maps and ordered dicts need to be merged when they're a child.

Reply:
I have made the changes as per our discussion:)